### PR TITLE
Stop emitting useless sig annotations in reflection.rbi

### LINKED ISSUE #3674: Hidden definitions cannot be generated for methods with more than one anonymous variable
Methods with more than one anonymous/unnamed variable generate invalid method definitions in hidden definitions. For example, if you include the `statsd-instrument` gem in your project, there are issues:

Statsd-instrument generates a number of methods with unnamed parameters, see https://github.com/Shopify/statsd-instrument/blob/2772e56f10ca614238817167f10be86ff4a4bbb6/lib/statsd/instrument/datagram_builder.rb#L22.

This causes the hidden definitions generation to put this in `reflection.rbi`:
```ruby
class StatsD::Instrument::StatsDDatagramBuilder < StatsD::Instrument::DatagramBuilder
  sig do
    params(
      _: ::T.untyped,
      _: ::T.untyped,
      _: ::T.untyped,
      _: ::T.untyped,
    )
    .returns(::T.untyped)
  end
  def d(_, _, _, _); end

  sig do
    params(
      _: ::T.untyped,
      _: ::T.untyped,
      _: ::T.untyped,
      _: ::T.untyped,
    )
    .returns(::T.untyped)
  end
  def h(_, _, _, _); end

  sig do
    params(
      _: ::T.untyped,
      _: ::T.untyped,
      _: ::T.untyped,
      _: ::T.untyped,
    )
    .returns(::T.untyped)
  end
  def kv(_, _, _, _); end

end
```

This then causes a bunch of the following errors in `reflection.json.err`:
```
/tmp/d20201117-13-f33zr2/reflection.rbi:753996: Hash key `_` is duplicated https://srb.help/3011
      753996 |      _: ::T.untyped,
                    ^
    /tmp/d20201117-13-f33zr2/reflection.rbi:753994: First occurrence of `_` hash key
      753994 |      _: ::T.untyped,
                    ^

/tmp/d20201117-13-f33zr2/reflection.rbi:753997: Hash key `_` is duplicated https://srb.help/3011
      753997 |      _: ::T.untyped,
                    ^
    /tmp/d20201117-13-f33zr2/reflection.rbi:753994: First occurrence of `_` hash key
      753994 |      _: ::T.untyped,
```

This issue with `statsd-instrument` as well as related issue #3603 with `aws-sdk-core` has meant we've had to lock our project to sorbet 0.5.5944, which is the last known release without these issues. It's broken in 0.5.6001, so the issues must've been introduced between those two versions.

### PR DESCRIPTION
**Pull request by jez**
### Motivation

Fixes #3674

For context see https://github.com/sorbet/sorbet/issues/3674#issuecomment-728674364

### Test plan

Added failing test, verified that it failed before the fix.

### CODE REVIEW

File: gems/sorbet/test/snapshot/partial/underscores/src/foo.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+class A
+  # This one tests hidden-definitions working
+  define_method(:foo) do |_, _|
+    puts 'hello, world!'
+  end
```
Comment (jez) [inline gems/sorbet/test/snapshot/partial/underscores/src/foo.rb:6]: hidden definitions uses a completely different codepath, and so it was working already. but I figured I'd add a test anyway

File: gems/sorbet/lib/serialize.rb [hunk 2]
```diff
@@ -174,7 +174,7 @@ def class_or_module(class_name)
       end
       next if denylisted_method(method)
       next if ancestor_has_method(method, Sorbet::Private::RealStdlib.real_singleton_class(klass))
-      serialize_method(method, true)
+      serialize_method(method, true, with_sig: false)
```
Comment (froydnj) [inline gems/sorbet/lib/serialize.rb:177]: If grepping is correct, it looks like these changes to pass `with_sig: false` make all the callsites pass `false`...is there any point in keeping the parameter and attendant code around?

**Review by froydnj — APPROVED** [review]
:stamp:

File: gems/sorbet/lib/serialize.rb (hunk 2 above)
Comment (jez) [reply gems/sorbet/lib/serialize.rb:177]: Nope, there's no point. We could do that as a follow up. In fact I actually made that change 9 months ago in Stripe's internal copy of this file:

http://go/ps/commit/85eb538af75

**Pull request merged**